src/compiler/checker.ts

```diff
@@ -566,7 +566,7 @@ module ts {
         }
 
         function getTargetOfExportAssignment(node: ExportAssignment): Symbol {
-            return resolveEntityName(<Identifier>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace);
+            return node.expression && resolveEntityName(<Identifier>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace);

```

Comment:
We treat identifier specially, as it means to export all meanings of the identifier, where as an exporting an expression, only exports the value. the binder handles these differently, if it is an identifier it will bind the symbol for the default export. so we know if we only have an identifier here.